github/actions_hosted_runners.go

```diff
@@ -0,0 +1,333 @@
+// Copyright 2020 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// HostedRunnerPublicIP represents the details of a public IP for github-hosted runner.
+type HostedRunnerPublicIP struct {
+	Enabled bool   `json:"enabled"`
+	Prefix  string `json:"prefix"`
+	Length  int    `json:"length"`
+}
+
+// HostedRunnerMachineSpec represents the details of a particular machine specification for github-hosted runner.
+type HostedRunnerMachineSpec struct {
+	ID        string `json:"id"`
+	CPUCores  int    `json:"cpu_cores"`
+	MemoryGB  int    `json:"memory_gb"`
+	StorageGB int    `json:"storage_gb"`
+}
+
+// HostedRunner represents a single github-hosted runner with additional details.
+type HostedRunner struct {
+	ID                 *int64                   `json:"id,omitempty"`
+	Name               *string                  `json:"name,omitempty"`
+	RunnerGroupID      *int64                   `json:"runner_group_id,omitempty"`
+	Platform           *string                  `json:"platform,omitempty"`
+	Image              *HostedRunnerImageDetail `json:"image,omitempty"`
+	MachineSizeDetails *HostedRunnerMachineSpec `json:"machine_size_details,omitempty"`
+	Status             *string                  `json:"status,omitempty"`
+	MaximumRunners     *int64                   `json:"maximum_runners,omitempty"`
+	PublicIPEnabled    *bool                    `json:"public_ip_enabled,omitempty"`
+	PublicIPs          []*HostedRunnerPublicIP  `json:"public_ips,omitempty"`
+	LastActiveOn       *string                  `json:"last_active_on,omitempty"`
+}
+
+// HostedRunnerImageDetail represents the image details of a github-hosted runners.
```

Comment:
In all comments, we try to use their official capitalization.

```suggestion
// HostedRunnerImageDetail represents the image details of a GitHub-hosted runners.
```